src/main/java/com/bettercloud/vault/VaultConfig.java

```diff
@@ -207,6 +208,57 @@ public VaultConfig readTimeout(final Integer readTimeout) {
         return this;
     }
 
+    /**
+     * <p>Set the "path depth" of the prefix path.  Normally this is just 
+     * 1, to correspond to one path element in the prefix path.  To use
+     * a longer prefix path, set this value
+     *
+     * @param prefixPathDepth integer number of path elements in the prefix path
+     */
+    public VaultConfig prefixPathDepth(int pathLength) {
+       if (pathLength < 1) {
+          throw new IllegalArgumentException("pathLength must be > 1");
+       }
+
+       this.prefixPathDepth = pathLength;
+       return this;
+    }
+
+
+    /**
+     * <p>Set the "path depth" of the prefix path, by explicitly specifying
+     * the prefix path, e.g., "foo/bar/blah" would set the prefix path depth
+     * to 3.
+     *
+     * @param prefixPath string prefix path, with or without initial or
+     * final forward slashes
+     */
+    public VaultConfig prefixPath(String prefixPath) {
```

Comment:
I suggest `setPrefixPathDepthFromString` because this method does not actually set a `prefixPath` attribute